File: C++/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;
        vector<int> count(26, 0);
        
        for (int i = 0; i < s.size(); ++i) {
            count[(int)s[i] - 97]++;
            count[(int)t[i] - 97]--;
        }
        
        for (int c : count)
            if (c != 0)
                return false;
        
        return true;
    }
    
    
    // bool isAnagram(string s, string t) {
    //     sort(s.begin(), s.end());
    //     sort(t.begin(), t.end());
    //     return s == t;
    // }
};

File: Python/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        sCharCount = {}
        tCharCount = {}

        for ch in s:
            if ch not in sCharCount:
                sCharCount[ch] = 0

            sCharCount[ch] += 1

        for ch in t:
            if ch not in tCharCount:
                tCharCount[ch] = 0

            tCharCount[ch] += 1

        for ch in sCharCount:
            if ch not in tCharCount or sCharCount[ch] != tCharCount[ch]:
                return False

        return len(sCharCount) == len(tCharCount)